lib/KBox/src/pages/NavigationPage.cpp

```diff
@@ -0,0 +1,280 @@
+    //DEBUG("APHeadingmode is: - %d", apHeadingMode);
+    //DEBUG("APDodgemode is: - %d", apDodgeMode);
+    return true;
+}
+
+bool NavigationPage::processEvent(const EncoderEvent &ee) {
+    encoderClicks = ee.rotation;
+    navEncoderClicks = navEncoderClicks + encoderClicks;
+    encoderClicks = 0;
+    return true;
+}
+
+String NavigationPage::formatMeasurement(float measure, const char *unit) {
+  // extra spaces at the end needed to clear previous value completely
+  // (we use a non-fixed width font)
+  char s[10];
+  snprintf(s, sizeof(s), "%.0f %s  ", measure, unit); //do not display decimals for headings
+  return String(s);
+}
+
+
+void NavigationPage::processMessage(const KMessage &message) {
+  message.accept(*this);
+}
+
+void NavigationPage::visit(const APMessage &ap) {
+    navTargetRudderPosition = ap.getTargetRudderPosition();
+    navRudderCommandSent = ap.getRudderCommandSent();
+}
+
+void NavigationPage::visit(const RUDMessage &rm) {
+    navRudderSensorPosition = rm.getRudderDisplayAngle();
+}
+
+void NavigationPage::visit(const IMUMessage &imu) {
+    if (imu.getCalibration() == 3){
+        imuCalibrated = true;
+    }
+    if (imuCalibrated){
+        if (!apMode){
+            apModeString = "APMode: Off   ";
+            apModeDisplay->setText(apModeString);
+            apModeDisplay->setColor(ColorGreen);
+        }
+    } else {
+        apModeString = "Calibrating   ";
+        apModeDisplay->setText(apModeString);
+        apModeDisplay->setColor(ColorRed);
+    }
+    
+    rawHeading = RadToDeg(imu.getCourse());
+    if (rawHeading < 0){
+        navCurrentHeading = 720 + rawHeading;//adding 720 for nav math purposes
+        navCurrentDisplayHeading = 360 + rawHeading;
+    } else {
+        navCurrentHeading = 720 + rawHeading;
+        navCurrentDisplayHeading =  rawHeading;
+    }
+    headingDisplay->setText(formatMeasurement(navCurrentDisplayHeading, ""));
+    headingDisplay->setColor(colorForCourse(navCurrentDisplayHeading));
+    
+    if (apMode == true && apHeadingMode == true ){
+        if (isTargetSet == false){
+            initialTargetHeading = navCurrentHeading;
+            navTargetHeading = initialTargetHeading;
+            isTargetSet = true;
+        } else {
+            navTargetHeading = initialTargetHeading + navEncoderClicks;
+            if ((navTargetHeading - 720) < 0){
+                navTargetDisplayHeading = 360 + (navTargetHeading - 720) ;
+            } else {
+                navTargetDisplayHeading = navTargetHeading - 720;
+            }
+        }
+        targetHeadingDisplay->setText(formatMeasurement(navTargetDisplayHeading, ""));
+        targetHeadingDisplay->setColor(colorForCourse(navTargetDisplayHeading));
+        
+        //following code recalculates navCurrentHeading based on "offcourse" value to avoid non-linear heading numbers across 180 degrees
+        navOffCourse = navCurrentDisplayHeading - navTargetDisplayHeading;
```

Comment:
It looks like `navOffCourse` is used only in this function so it does not need to be declared in the class. You can just declare it in this function.